sirepo/server.py

```diff
@@ -140,7 +142,9 @@ def api_downloadFile(self, simulation_type, simulation_id, filename):
                 sirepo.util.raise_not_found("lib_file={} not found", p)
             raise
 
-    @sirepo.api.Spec("allow_visitor")
+    @sirepo.api.Spec(
+        "allow_visitor", error="String all_input", client_ip="IPAddress optional"
```

Comment:
spec=ErrorLoggingSpec (no need for all_input). I think we don't need client_ip here. What we'll do is ErrorLoggingSpec will know that it has to extract client_ip from the request.

>